java/arcs/core/data/ParticleSpec.kt

```diff
@@ -22,5 +22,13 @@ data class ParticleSpec(
     val claims: List<Claim> = emptyList(),
     val checks: List<Check> = emptyList(),
     /** Indicates whether the particle is an isolated (non-egress) particle. */
-    val isolated: Boolean = false
-)
+    val isolated: Boolean = false,
+    /** Optional egress type of the particle. Must be null for isolated particles. */
```

Comment:
Should we add an enum or a data class to capture the particle kind (if that is a concept)? That way we don't have to maintain the consistency between fields with `require` checks?

Reply:
At the moment we only have egress and isolated, but we keep talking about adding ingress. It's not clear yet whether those are mutually exclusive, but I would imagine not, in which case it could be isolated, egress, ingress, or ingress+egress.

Not really sure what the best way to represent that is. Might just wait until we actually need it.